wordpress-mobile/WordPress-Android
# Issue/4728 show quick actions in notifications only if device is unlocked

Fixes #4728 

This PR actually works for all notifications that have quick actions (that is, Comment notifications and the specific case of issue #4278 which is the 2FA push notification).
What it does is it re-creates the notification to only show quick actions if the device is unlocked and the app is unlocked as well.
The app now registers a broadcast receiver to check when the screen goes off, or when the screen is unlocked, and also checks the case that the user sets a PIN lock within WPAndroid app and still has a notification in the system dashboard.

To test:
CASE A: 2FA
- enable 2FA on your website
- log in to your app, then lock your phone's screen once logged in
- log in to your website on a desktop browser
- the 2FA auth push is received, with NO quick actions available.
- unlock the phone screen
- you should now see the quick actions APPROVE/IGNORE available in the push notification.
- If you lock the screen again, you should see the quick actions disappear from such notification (and viceversa, as many times as you do this).
- Tap "Approve" to log in to your site

CASE B: Comments
- log in to your app, then lock your phone's screen once logged in
- have some other account like a post of a blog you're an admin of
- the comment notification is received, with NO quick actions available.
- unlock the phone screen
- you should now see the quick actions REPLY/LIKE or REPLY/APPROVE (depending on your moderation settings for such blog) available.
- If you lock the screen again, you should see the quick actions disappear from such notification (and viceversa, as many times as you do this).

cc @hypest @kwonye @daniloercoli  
assigning to @hypest as he had the quick actions for 2FA notif idea originally :)
